### PR TITLE
Change sidebar tree creation to use path instead of parentId. Also move sidebar to web-ui

### PR DESCRIPTION
**Pull request by andresgutgon**
# What?
We decided to start using [document.path](https://github.com/latitude-dev/latitude-llm/pull/45) as source of truth. We need to change sidebar tree creation

### CODE REVIEW

File: packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    name?: string
+  }) {
+    this.id = id
+    this.name = isRoot ? 'root' : name
+    this.isRoot = isRoot
+    this.children = children
+    this.doc = doc
+  }
+
+  // Useful for testing
```
Comment (geclos) [inline packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts:35]: no need for the comment

File: packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  public doc?: SidebarDocument
+
+  constructor({
+    id,
+    doc,
+    children = [],
+    isRoot = false,
+    name = '',
+  }: {
+    id: string
```
Comment (geclos) [inline packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts:22]: what is this id? do we need it?

File: packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  generateNodeId = defaultGenerateNodeUuid,
+}: {
+  documents: SidebarDocument[]
+  generateNodeId?: typeof defaultGenerateNodeUuid
+}) {
+  return useMemo(() => {
+    const root = new Node({ id: generateNodeId(), children: [], isRoot: true })
+    const nodeMap = new Map<string, Node>()
+    nodeMap.set('', root)
+    const sorted = documents.slice().sort(sortByPathDepth)
```
Comment (geclos) [inline packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts:113]: minor gripe but I'm sure you can do this without this pre sorting step

File: packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts (hunk 2 above)
Comment (andresgutgon) [reply packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts:22]: for `key` react elements. Is nice to do `key={node.id}`

File: packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        })
+        nodeMap.set(path, node)
+
+        const parentPath = path.split('/').slice(0, -1).join('/')
+
+        // We force TypeScript to check that the parentPath is not empty
+        // We pre-sorted documents by path depth, so we know
+        // that the parent node exists
+        const parent = nodeMap.get(parentPath)!
+        parent.children.push(node)
```
Comment (andresgutgon) [inline packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts:93]: Before pushing the node find their place alphabetically. if it's a folder goes first.

**Review by csansoon — APPROVED** [review]
👍

File: packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts (hunk 4 above)
Comment (andresgutgon) [reply packages/web-ui/src/sections/DocumentsSidebar/useTree/index.ts:93]: I'll do this in another PR Carlos wants this code

**Pull request merged**